packages/nimble-components/CONTRIBUTING.md

```diff
@@ -77,15 +77,33 @@ Before building a new component, 3 specification documents need to be created:
 
 ## Develop new components
 
-### Marking a component as in development
+### Marking a component as incubating
 
-If a component will require multiple pull requests before having a complete and stable API, it should be marked as "in-development" to indicate to clients that they shouldn't start using it yet. To do this:
+If a component is not ready for general use, it should be marked as "incubating" to indicate that status to clients. A component could be in this state if any of the following are true:
+
+-   It is still in development.
+-   It is currently experimental or application-specific and hasn't yet been generalized for broader use.
+-   It is missing important features like interaction design, visual design, accessibility, or framework integration.
+
+Incubating contributions may compromise on the above capabilities but they still must abide by other repository requirements. For example:
+
+-   Coding conventions (element naming, linting, code quality)
+-   Unit and Chromatic test coverage
+-   Storybook documentation
+
+To mark a component as incubating:
```

Comment:
@rajsite I updated based on our conversation this morning.

When this completes I'm happy to update our existing incubating components (table, wafer map, tooltip) to follow this guidance and terminology. Do you think we should make the path change? (It would be breaking).

Reply:
Actually... maybe not the table? That just might be disruptive. And it would be a really big move (it's dozens of components).

Reply:
I'd leave the table but move wafer map and tooltip for now.
But new external experimental table columns can start the pattern, ie incubating/table-column/select

Reply:
@rajsite I've stewed on including `incubating` in the path for components (and the entry point for Angular integration and the namespace for Blazor integration) and I'm finding myself uncomfortable with the idea. I think my reservations are:
1. it takes work to move something when it is done incubating (as you mention, it'd be disruptive to do for the table right now)
2. it definitely impacts clients who have uptaken the incubating component. "The component you're already using is officially official. To celebrate, your app is broken 🎉"
3. it might impact clients who haven't uptaken the incubating component (see [this PR](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/494699) fixing breakages from the Angular entry point change in apps that were using sl-lib but not the table and tried to uptake the new nimble)
4. we have to come up with a bunch of decisions around exactly where "incubating" appears in paths/entry point names/namespaces, document those decisions, etc

The main benefit would be that clients would be less likely to accidentally use an incubating component and think it's official. But I think we get a lot of that benefit from the Storybook organization change and that clients who ignore the warnings in our docs are likely to ignore the signal we send in the component path too. So that benefit isn't justifying the reservations I list above.

In the interest of pushing this HLD forward next week, what do you say we shelve this idea for now? We can always introduce it in a future pass if we refine our idea of incubating components as we add more of them.

Reply:
I'm sold on dropping the source location / namespace change and relying on documentation. Just to confirm you mean drop it altogether right? So removing the following:

https://github.com/ni/nimble/blob/11f75048028d607e179a5e9d7129987b2bce81e8/packages/nimble-components/CONTRIBUTING.md?plain=1#L100

Reply:
Correct! I'll make that change.